napari/_tests/test_advanced.py

```diff
@@ -298,3 +339,39 @@ def test_adding_removing_layer(make_test_viewer):
     viewer.layers.append(layer)
     assert len(viewer.layers) == 1
     assert viewer.dims.ndim == 4
+
+
+@pytest.mark.parametrize('Layer, data, ndim', layer_test_data)
+def test_add_remove_layer_external_callbacks(
+    make_test_viewer, Layer, data, ndim
+):
+    """Test external callbacks for layer emmitters preserved."""
+    viewer = make_test_viewer()
+
+    layer = Layer(data)
+    # Check layer has been correctly created
+    assert layer.ndim == ndim
+
+    # Connect a custom callback
+    def my_custom_callback(event):
+        return
+
+    layer.events.connect(my_custom_callback)
+
+    # Check that no internal callbacks have been registered
+    len(layer.events.callbacks) == 1
+    for em in layer.events.emitters.values():
+        assert len(em.callbacks) == 1
+
+    viewer.layers.append(layer)
+    # Check layer added correctly
+    assert len(viewer.layers) == 1
+
```

Comment:
Maybe here add a check that callbacks are longer now?

```suggestion

# check that adding a layer created new callbacks
assert any(len(em.callbacks) > 1 for em in layer.events.emitters.values())

```

Reply:
done